File: Leetcode/tree/111_MinimumDepthOfBinaryTree/Solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root is None:
            return 0
        if not root.left and not root.right:
            return 1
        if root.left is None:
            return 1+self.minDepth(root.right)
        if root.right is None:
            return 1+self.minDepth(root.left)
        return min(1+self.minDepth(root.left), 1+self.minDepth(root.right))

File: Leetcode/tree/111_MinimumDepthOfBinaryTree/Solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if(root==nullptr)
            return 0;
        if(root->left==nullptr && root->right==nullptr)
            return 1;
        return (root->left==nullptr)? 1+minDepth(root->right):
                (root->right==nullptr)? 1+minDepth(root->left):
                min(1+minDepth(root->left), 1+minDepth(root->right));
    }
};
